cmd/anonymizer/app/anonymizer/anonymizer.go

```diff
@@ -166,3 +204,43 @@ func filterTags(tags []model.KeyValue) []model.KeyValue {
 	}
 	return out
 }
+
+// filterCustomTags returns all tags other than allowedTags
+func filterCustomTags(tags []model.KeyValue) []model.KeyValue {
+	out := make([]model.KeyValue, 0, len(tags))
+	for _, tag := range tags {
+		if !allowedTags[tag.Key] {
+			out = append(out, tag)
+		}
+	}
+	return out
+}
+
+// hashTags converts each tag into corresponding string values
+// and then find its hash
+func hashTags(tags []model.KeyValue) []model.KeyValue {
+	var vStr string
+	var kv model.KeyValue
+	hTags := make([]model.KeyValue, 0, len(tags))
+
+	for _, tag := range tags {
+		switch tag.VType {
+		case model.BinaryType:
+			vStr = string(tag.VBinary)
+		case model.BoolType:
+			vStr = strconv.FormatBool(tag.VBool)
+		case model.StringType:
+			vStr = tag.VStr
+		case model.Int64Type:
+			vStr = strconv.FormatInt(tag.VInt64, 10)
+		case model.Float64Type:
+			vStr = strconv.FormatFloat(tag.VFloat64, 'E', -1, 64)
+		}
+
+		kv.VType = model.StringType
+		kv.VStr = hash(vStr)
```

Comment:
instead of reusing variable and manipulating fields, use `model.String(key, value)` function

https://github.com/jaegertracing/jaeger/blob/9f954fe7f022a0c9fb9aeee378a912958c8c3094/model/keyvalue.go#L45